src/shared/components/table-grid/Table.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+
+import csx from './Table.scss';
+
+namespace Table {
+  export interface Cell {
+    value?: string | number;
+    component?: JSX.Element;
+    className?: string;
+  }
+
+  export interface Row {
+    [key: string]: Cell;
+  }
+
+  export interface Props {
+    data: Row[];
+    header: string[];
+  }
+}
+
+const Table = ({ data, header }: Table.Props) => {
+  return (
+    <div style={{
+      gridTemplateColumns: `repeat(${header.length}, ${header.length}fr)`
+    }} className={csx.grid}>
+      {header.map((key, i) => (
+        <span className={csx.header} key={i}>
+          {key}
+        </span>
+      ))}
+      {data.map((entry, index) =>
+        Object.values(entry).map(
```

Comment:
use filter before map to filterout all falsy components - then you can avoid cell.component && inside map. Also this should be put to function which is memoized or smth like that because Object.values is called on every re-render